### PR TITLE
Add chart release process

### PR DESCRIPTION
**Pull request by michaellzc**
Since we're no longer doing one commit per release, it makes sense to add a (temporary) release process.

Also, added a CI check to verify helm-docs are up-to-date

## Test plan

- Make a change to README.md.gotmpl without updating helm-docs
- Commit and watch for buildkite build

Sample
- https://buildkite.com/sourcegraph/deploy-sourcegraph-helm/builds/184
- [`c74bbea` (#55)](https://github.com/sourcegraph/deploy-sourcegraph-helm/pull/55/commits/c74bbead1bcd0fd2813dda06c09f35eefbbcd734)

close https://github.com/sourcegraph/sourcegraph/issues/32016

### CODE REVIEW

**Comment by michaellzc** [comment]
Current dependencies on/for this PR:
* main
  * **PR #55** <a href="https://app.graphite.dev/github/pr/sourcegraph/deploy-sourcegraph-helm/55" target="_blank">Graphite</a>  (this PR)

This comment was auto-generated by [Graphite](https://graphite.dev?utm_source=gh).

File: .buildkite/pipeline.yaml [hunk 1]
```diff
@@ -18,3 +18,10 @@ steps:
     soft_fail:
       - exit_status: 255
     agents: { queue: job }
+  
+  - label: ":book: Vericy helm-docs is up-to-date"
```
Comment (caugustus-sourcegraph) [inline .buildkite/pipeline.yaml:22]:
```suggestion
  - label: ":book: Verify helm-docs is up-to-date"
```

File: .buildkite/pipeline.yaml [hunk 2]
```diff
@@ -18,3 +18,10 @@ steps:
...
+    commands:
+      - "./scripts/helm-docs.sh"
+      - "echo \"checking for uncommiteed changes\""
```
Comment (caugustus-sourcegraph) [inline .buildkite/pipeline.yaml:25]:
```suggestion
      - "echo \"checking for uncommitted changes\""
```

File: RELEASE.md [hunk 3]
```diff
@@ -0,0 +1,27 @@
+# Sourcegraph Helm Chart Release
+
+> Sourcegraph helm chart may be released outside the regular [Sourcegraph releases] schedule, these steps are required whenever releaseing a new version.
+
+## 1. Bump the `version` in `Chart.yaml`
+
+Doulbe check the `CHANGELOG.md` see if there is any `**BREAKING**:` changes and follow [semver] to bump chart `version` accordingly.
```
Comment (caugustus-sourcegraph) [inline RELEASE.md:7]:
```suggestion
Double check the `CHANGELOG.md` to see if there are any `**BREAKING**:` changes and follow [semver] to bump chart `version` accordingly. Increase the minor version for a breaking change or Sourcegraph upgrade, patch version for a backwards-compatible change.
```

Comment (caugustus-sourcegraph) [reply RELEASE.md:7]: Sorry, that started out as a typo fix and then I figured "why not explain more". Feel free to ignore!

**Review by caugustus-sourcegraph — APPROVED** [review]
Thank you!

**Pull request merged**